legend-engine-services-model-api/src/main/java/org/finos/legend/engine/plan/execution/service/api/ServiceModelingApi.java

```diff
@@ -95,9 +93,16 @@ public Response doTest(PureModelContext service, @ApiParam(hidden = true) @Pac4J
         }
         catch (Exception ex)
         {
+            String servicePattern = null;
+            if (service instanceof PureModelContextData)
+            {
+                PureModelContextData data = ((PureModelContextData) service).shallowCopy();
+                Service invokedService = (Service) Iterate.detect(data.getElements(), e -> e instanceof Service);
+                servicePattern = invokedService == null ? null : invokedService.pattern;
+            }
             MetricsHandler.observe("service test error", start, System.currentTimeMillis());
```

Comment:
Do you intend to remove the call to observe as well ?

Reply:
Yes, this should probably be removed in this case